controllers/bootstrap/init.go

```diff
@@ -497,13 +531,53 @@ func (b *Bootstrap) waitOperatorReady(name, namespace string) error {
 				klog.Errorf("Failed to get Cluster Service Version: %v", err)
 				return false, err
 			}
-			if csv.Status.Phase == olmv1alpha1.CSVPhaseSucceeded {
-				return true, nil
+			if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded {
+				return false, nil
+			}
+			if csv.Status.Reason != olmv1alpha1.CSVReasonInstallSuccessful {
+				return false, nil
 			}
+			klog.Errorf("Cluster Service Version %s/%s is ready", csv.Namespace, csv.Name)
```

Comment:
Should we use `klog.infof` instead of `errorf`?

Reply:
Nice catch, updated